packages/concerto-analysis/src/comparers/class-declarations.ts

```diff
@@ -54,11 +54,17 @@ const classDeclarationTypeChanged: ComparerFactory = (context) => ({
         if (aType === bType) {
             return;
         }
-        context.report({
-            key: 'class-declaration-type-changed',
-            message: `The ${aType} "${a.getName()}" changed type from ${aType} to ${bType}`,
-            element: a
-        });
+
+        const isAbstract = (declaration) => declaration.isAbstract();
+        if (isAbstract(a) !== isAbstract(b)) {
+            const changeType = isAbstract(a) ? 'abstract to concrete' : 'concrete to abstract';
+            const changeSeverity = isAbstract(a) ? 'minor' : 'major';
+            context.report({
+                key: 'class-declaration-abstractness-changed',
+                message: `The class "${a.getName()}" changed from ${changeType} (${changeSeverity} change).`,
```

Comment:
The severity can't simply be expressed in the message but must be defined here:
https://github.com/accordproject/concerto/blob/85366be18b5deb3efaa6acfbc7319b14a0219c05/packages/concerto-analysis/src/compare-config.ts#L48

You'll need to create two different keys: one for abstract to concrete, and one for concrete to abstract.

Reply:
I have added the rules, and change the severity to changeKey.
The pull request is ready for review.
let me know if any changes need.